storage/statedb/database.go

```diff
@@ -751,12 +761,14 @@ func (db *Database) Cap(limit common.StorageSize) error {
 	// If the preimage cache got large enough, push to disk. If it's still small
 	// leave for later to deduplicate writes.
 	flushPreimages := db.preimagesSize > 4*1024*1024
+	numPreimages := 0
 	if flushPreimages {
-		if err := db.writeBatchPreimages(); err != nil {
-			db.lock.RUnlock()
-			return err
-		}
+		db.diskDB.WritePreimages(0, db.preimages)
+		numPreimages = len(db.preimages)
 	}
+	db.diskDB.WritePruningMarks(db.pruningMarks)
```

Comment:
`(64B + 39B) * # of dirty nodes` are continuously and unconditionally stored in MiscDB. 
Good for future use of pruning becuase these provide the exact obsolete node hash and delete them after lookup.
Bad for uncertain about future-use of pruning becauase these are continuously take disk space. The [provided numbers](https://github.com/klaytn/klaytn/pull/1871#issuecomment-1641223838) seem not to show this disk usage. How do you anticipate it?

Reply:
When live pruning is disabled, no pruning marks are written. https://github.com/klaytn/klaytn/blob/320554640a13fbadf12d42db0a0f91c0edc14386/storage/statedb/trie.go#L605-L611